packages/@uppy/companion/src/companion.js

```diff
@@ -212,25 +212,33 @@ const getOptionsMiddleware = (options) => {
   if (options.providerOptions.s3) {
     const S3 = require('aws-sdk/clients/s3')
     const AWS = require('aws-sdk')
-    const config = options.providerOptions.s3
+    const s3ProviderOptions = options.providerOptions.s3
+    const s3Options = Object.assign({
+      signatureVersion: 'v4'
+    }, s3ProviderOptions)
+    // delete options intended for Companion, not S3:
+    delete s3Options.key
+    delete s3Options.secret
+    delete s3Options.conditions
+    delete s3Options.getKey
+
+    if (s3ProviderOptions.accessKeyId || s3ProviderOptions.secretAccessKey) {
```

Comment:
should we support both instead? 🤔 So when `accessKeyId` and `secretAccessKey` are supplied, we set the values to `key` and `secret`?

Reply:
i think we should support just one way to do it—else we'd still need a check for what happens when you provide both an `accessKeyId` and a `key`